pallets/creditcoin/src/lib.rs

```diff
@@ -543,55 +551,68 @@ pub mod pallet {
 				KillStorageResult::AllRemoved(u) => u,
 			};
 
-			let mut deals_count = 0usize;
+			let mut deals_count = 0u32;
 			let deals_to_keep: Vec<_> = DealOrders::<T>::drain_prefix(block_number)
 				.filter_map(|(hash, deal)| {
-					deals_count += 1;
+					deals_count = deals_count.saturating_add(1);
 					if deal.funding_transfer_id.is_some() {
 						Some((DealOrderId::with_expiration_hash::<T>(block_number, hash), deal))
 					} else {
 						None
 					}
 				})
 				.collect();
-			let funded_deals_count = deals_to_keep.len();
-			let deals_count = deals_count - funded_deals_count;
+			let funded_deals_count = deals_to_keep.len().unique_saturated_into();
+			let deals_count = deals_count.saturating_sub(funded_deals_count);
 			for (key, deal) in deals_to_keep {
 				DealOrders::<T>::insert_id(key, deal);
 			}
 
 			<T as Config>::WeightInfo::on_initialize(
-				0,
 				ask_count,
 				bid_count,
 				offer_count,
-				deals_count.unique_saturated_into(),
-				funded_deals_count.unique_saturated_into(),
+				deals_count,
+				funded_deals_count,
+				unverified_count,
 			)
 		}
 
 		fn offchain_worker(_block_number: T::BlockNumber) {
 			if let Some(auth_id) = Self::authority_id() {
 				let auth_id = T::FromAccountId::from(auth_id);
-				for pending in UnverifiedTransfers::<T>::get() {
+				for (deadline, transfer_id, pending) in UnverifiedTransfers::<T>::iter() {
+					let storage_key =
+						ocw::transfer_local_status_storage_key::<T>(deadline, &transfer_id);
+					let transfer_status = ocw::LocalVerificationStatus::new(&storage_key);
+					if transfer_status.is_complete() {
+						log::debug!("Already handled transfer ({:?}, {:?})", deadline, transfer_id);
+						continue;
```

Comment:
I'm not quire sure how the `deadline` value is taken into account in the decision whether we should verify this transfer or not. But I think the existing offchain worker tests don't simulate a scenario where the deadline has expired.

Reply:
Here the `deadline` is just used in combination with the transfer ID to identify this specific attempt to register the transfer (so this way if someone resubmits the transfer later we will retry it because the deadline is different). The deadline only really matters so we don't retry transfers infinitely, the OCW logic doesn't take it into account.

Reply:
OK, I understand the logic now. 

However the coverage reports show that there are still some lines not being exercised by tests. Is it going to be easy to add some tests for them or should I try to do that at a later stage ?